### PR TITLE
Add drools checker

### PR DESCRIPTION
**Pull request by Martin-Idel-SI**
This PR introduces a checker workflow step.

Current scope:
- Rules can be written in Drools and have to be provided in a folder referenced in the workflow step
- Rules will be evaluated for all artifacts and the build will fail if any rule fires

Missing:
- Documentation (the API may still change significantly, until then, the example projects will show how to use the features)

Features to be added:
- Allow specifying evaluation results (currently, all rule failures will be added to a "dummy" policy evaluation.
- Allow or simplify writing rules comparing information between artifacts.

### CODE REVIEW

File: example-projects/example-project/src/policies/rules/DummyRule.drl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.eclipse.sw360.antenna.exampleproject
```
Comment (maxhbr) [inline example-projects/example-project/src/policies/rules/DummyRule.drl]: Missing License Header

**Comment by Martin-Idel-SI** [comment]
@maxhbr Done. Please review again.

File: antenna-testing/antenna-frontend-stubs-testing/src/main/java/org/eclipse/sw360/antenna/frontend/testProjects/ExampleTestProject.java [hunk 2]
```diff
@@ -31,7 +31,8 @@ public ExampleTestProject() {
 
     @Override
     public List<String> getOtherFilesToCopy() {
-        return Stream.of("src/reportData.json", "src/dependencies.csv").collect(Collectors.toList());
+        return Stream.of("src/reportData.json", "src/dependencies.csv", "src/policies/rules/DummyRule.drl", "src/policies/policies.properties")
```
[the remark this answers is not in the document]
Comment (Martin-Idel-SI) [reply antenna-testing/antenna-frontend-stubs-testing/src/main/java/org/eclipse/sw360/antenna/frontend/testProjects/ExampleTestProject.java]: I'd say, that's to be decided.

I would assume that the rules will be managed centrally, but currently, they are provided by providing a path in the workflow.xml. Hence the folder must be copied to the temporary directory for testing.

File: antenna-workflow-steps/processors/checkers/antenna-drools-checker/src/main/java/org/eclipse/sw360/antenna/bundle/DroolsEngine.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    private List<File> getAllRulesInConfiguredFolder() {
+        try {
+            Path folder_path = Paths.get(rulesetDirectory, rulesetPath, RULES_SUBFOLDER);
+            return Files.walk(folder_path)
+                    .filter(Files::isRegularFile)
+                    .filter(path -> path.toString().endsWith(".drl"))
+                    .map(Path::toFile)
+                    .collect(Collectors.toList());
+        } catch (IOException ex) {
+            throw new AntennaExecutionException("Exception when trying to read file [" + ex.getMessage() + "]", ex);
```
[the remark this answers is not in the document]
Comment (Martin-Idel-SI) [reply antenna-workflow-steps/processors/checkers/antenna-drools-checker/src/main/java/org/eclipse/sw360/antenna/bundle/DroolsEngine.java]: I ditched Files.walk. The new implementation should not throw any IOException. At most, the ruleset will be empty and you can find some things in the log.

File: antenna-workflow-steps/processors/checkers/antenna-drools-checker/src/main/java/org/eclipse/sw360/antenna/bundle/DroolsEngine.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        Path policiesVersionPath = Paths.get(rulesetDirectory, rulesetPath, POLICIES_PROPERTIES_FILENAME);
+        try (FileInputStream policiesFile = new FileInputStream((policiesVersionPath.toFile()));
+             InputStream policiesStream = new DataInputStream(policiesFile)) {
+            appProperties.load(policiesStream);
+            return appProperties.getProperty(POLICIES_VERSION);
+        } catch (IOException ex) {
+            logger.error("Could not find or read version file. Details: " + ex.getMessage());
+        }
+
+        return "noVersion";
```
[the remark this answers is not in the document]
Comment (Martin-Idel-SI) [reply antenna-workflow-steps/processors/checkers/antenna-drools-checker/src/main/java/org/eclipse/sw360/antenna/bundle/DroolsEngine.java]: Using Optional.empty(). The magic string (slightly reworded) has moved to AntennaDroolsChecker. We could discuss throwing an exception if the version string is empty, but this must be specified.

**Comment by Martin-Idel-SI** [comment]
@maxhbr Done. I guess we're ready to go then.

**Pull request merged**